### PR TITLE
Only run CI on PRs and pushes to `main`

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,7 +1,13 @@
 name: doctest-CI
+
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
```
Comment (andreasabel) [inline .github/workflows/ci.yml:8]: I don't know whether restricting `pull_request` to `main` is necessary. But then, PRs are only made to `main` on this repo, so it looks good!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,7 +1,13 @@
 name: doctest-CI
+
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - main
```
Comment (andreasabel) [inline .github/workflows/ci.yml:6]: For Agda, we also have dedicated branches `ci-*` that run the CI.  Useful if you want feedback before or without opening a PR.

E.g. https://github.com/agda/agda/blob/dc56491ca9399ad4f55341e693308b44f354770c/src/github/workflows/stack.yml#L3-L20
(note that the `&`-YAML-syntax is not understood by github actions, so we expand it in a normalization step when constructing the actual `.github/workflows/stack.yml`).

**Review by andreasabel — APPROVED** [review]

**Comment by andreasabel** [comment]
Btw, I got the following alert from github:
> The default branch has been renamed!
>
> `master` is now named `main`
> 
> If you have a local clone, you can update it by running the following commands...

What is the motivation for renaming `master`?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sol) [reply .github/workflows/ci.yml:6]: Ah, that sounds good.  I was using "drapft" PRs for the very same purpose.

**Comment by sol** [comment]
@andreasabel `main` is the default branch name for GitHub and recent (upcoming?) versions of `git`. So I think it makes sense to go with that.  I'm not interested interested in discussing politics.

**Comment by andreasabel** [comment]
> @andreasabel `main` is the default branch name for GitHub and recent (upcoming?) versions of `git`. So I think it makes sense to go with that. I'm not interested interested in discussing politics.

My fingers automatically type `master`...

I have no understanding for these changes which just burn developer time.  I do not see any offence in the word "master" (unless maybe if contrasted with "slave").  Maybe I should start to call my master students "main students".  The bachelor students would then not be my main students.  Surely makes the world a fairer place, haha.

**Pull request merged**